### PR TITLE
Feature/bge lc  edit batch

### PR DESCRIPTION
**Pull request by bethbrains**
# Critical Changes

# Changes
- User can edit BGE Batch settings, sees warnings if records already exist.

# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by bethbrains** [comment]
**lurch: attach W-028758

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x80000007fOClAAM" target="_blank">W-028758</a>

File: src/labels/CustomLabels.labels [hunk 1]
```diff
@@ -2504,6 +2504,27 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
...
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Warning message for Field Options step of BGE Config Wizard</shortDescription>
+        <value>WARNING: There are records saved to this batch. Any changes you make won&apos;t affect existing records.</value>
+    </labels>
+    <labels>
+        <fullName>bgeEditBatchFieldSelectionWarning</fullName>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Warning msg for Select Fields step of BGE Config Wizard</shortDescription>
```
Comment (lmeerkatz) [inline src/labels/CustomLabels.labels]: msg -> message

Comment (bethbrains) [reply src/labels/CustomLabels.labels]: I was shortening to get under char limit, but in the end this shortening wasn't necessary. Thanks. :)

File: src/classes/BGE_ConfigurationWizard_CTRL.cls [hunk 2]
```diff
@@ -472,7 +487,7 @@ public with sharing class BGE_ConfigurationWizard_CTRL {
         @AuraEnabled public String description;
         @AuraEnabled public Id id;
         @AuraEnabled public Boolean requireTotalMatch;
-        @AuraEnabled public Integer expectedCount;
+        @AuraEnabled public Decimal expectedCount;
```
[the remark this answers is not in the document]
Comment (bethbrains) [reply src/classes/BGE_ConfigurationWizard_CTRL.cls]: fixed. we were getting an error on the other one and changed them both unnecessarily.

**Comment by bethbrains** [comment]
@kofijohnson ready for final review. changes as per laura's comments have been incorporated.

File: src/classes/BGE_ConfigurationWizard_CTRL.cls [hunk 3]
```diff
@@ -58,41 +58,56 @@ public with sharing class BGE_ConfigurationWizard_CTRL {
...
-                            'RequireTotalMatch__c, ' +
-                            'Active_Fields__c, ' +
-                            'Batch_Process_Size__c, ' +
-                            'Run_Opportunity_Rollups_while_Processing__c, ' +
-                            'Contact_Matching_Rule__c, ' +
-                            'Donation_Matching_Behavior__c, ' +
-                            'Donation_Matching_Rule__c, ' +
-                            'Donation_Date_Range__c, ' +
-                            'Process_Using_Scheduled_Job__c' +
+                    List<String> fields = new List<String>{
```
Comment (lmeerkatz) [inline src/classes/BGE_ConfigurationWizard_CTRL.cls]: You could use a set here to avoid duplicates in the future. #takeitorleaveit

Comment (randi274) [reply src/classes/BGE_ConfigurationWizard_CTRL.cls]: I'll take it!

**Comment by cpose** [comment]
QA Approved.

**Pull request merged**